geolocator_web/lib/src/html_geolocation_manager.dart

```
@@ -1,46 +1,77 @@
-import 'dart:html' as html;
+import 'dart:async';
+import 'dart:js_interop';
 
 import 'package:geolocator_platform_interface/geolocator_platform_interface.dart';
+import 'package:web/web.dart' as web;
 
 import 'geolocation_manager.dart';
 import 'utils.dart';
 
 /// Implementation of the [GeolocationManager] interface based on the
 /// [html.Geolocation] class.
 class HtmlGeolocationManager implements GeolocationManager {
-  final html.Geolocation _geolocation;
+  final web.Geolocation _geolocation;
 
   /// Creates a new instance of the [HtmlGeolocationManager] class.
-  HtmlGeolocationManager() : _geolocation = html.window.navigator.geolocation;
+  HtmlGeolocationManager() : _geolocation = web.window.navigator.geolocation;
 
   @override
   Future<Position> getCurrentPosition({
     bool? enableHighAccuracy,
     Duration? timeout,
   }) async {
+    Completer<Position> completer = Completer();
     try {
-      final geoPosition = await _geolocation.getCurrentPosition(
-        enableHighAccuracy: enableHighAccuracy,
-        timeout: timeout,
+      _geolocation.getCurrentPosition(
+        (web.GeolocationPosition position) {
+          completer.complete(toPosition(position));
+        }.toJS,
+        (web.GeolocationPositionError error) {
+          completer.completeError(convertPositionError(error));
+        }.toJS,
+        web.PositionOptions(
+          enableHighAccuracy: enableHighAccuracy ?? false,
+          timeout:
+              timeout?.inMicroseconds ?? const Duration(days: 1).inMilliseconds,
+        ),
       );
-
-      return toPosition(geoPosition);
-    } on html.PositionError catch (e) {
-      throw convertPositionError(e);
+    } catch (e) {
+      completer.completeError(e);
```

Please convert the error to one of the custom exceptions defined in the `convertPositionError` method before completing the completer. 

This will throw a known exception to the application developers so they don't have to update their application logic and handle unexpected exceptions solely for the geolocator_web plugin.

---

I've returned a platform exception so the caught exception can be passed to the detail field. Another option could be the `PositionUpdateException` exception

---

ok, my only concern there is the underlying exception will be lost. Perhaps we could add an optional cause field to the PositionUpdateException?